### PR TITLE
epicli upgrade - implementation

### PR DESCRIPTION
**Pull request by seriva**
#649

### CODE REVIEW

File: core/src/epicli/cli/engine/ansible/AnsibleRunner.py [hunk 1]
```diff
@@ -69,13 +68,62 @@ def run(self):
...
+        # pre-flight to prepare machines
+        self.pre_flight(inventory_path)
+
+        # run upgrade playbook
         self.ansible_command.run_playbook(inventory=inventory_path,
-                                          playbook_path=self.playbook_path('repository_teardown'))
+                                          playbook_path=self.playbook_path('upgrade'))
+
+        #post-flight after we are done
+        self.pre_flight(inventory_path)
```
Comment (to-bar) [inline core/src/epicli/cli/engine/ansible/AnsibleRunner.py]:
```suggestion
        self.post_flight(inventory_path)
```

File: core/src/epicli/cli/engine/ansible/AnsibleRunner.py [hunk 2]
```diff
@@ -69,13 +68,62 @@ def run(self):
...
+
+        # run roles
+        enabled_roles = inventory_creator.get_enabled_roles()
         for role in enabled_roles:
             self.ansible_command.run_playbook(inventory=inventory_path,
                                               playbook_path=self.playbook_path(to_role_name(role)))
 
+        #post-flight after we are done
+        self.pre_flight(inventory_path)
```
Comment (to-bar) [inline core/src/epicli/cli/engine/ansible/AnsibleRunner.py]:
```suggestion
        self.post_flight(inventory_path)
```

File: core/src/epicli/cli/epicli.py [hunk 3]
```diff
@@ -163,9 +163,11 @@ def run_delete(args):
 
 def upgrade_parser(subparsers):
     sub_parser = subparsers.add_parser('upgrade',
-                                       description='Upgrades the Docker and K8n roles of an existing Epiphany Platform to latest version.')
+                                       description='Upgrades the Docker and K8n roles of an existing Epiphany Platform to the latest version.')
```
Comment (to-bar) [inline core/src/epicli/cli/epicli.py]:
```suggestion
                                       description='Upgrades K8s components of an existing Epiphany Platform cluster.')
```

File: core/src/epicli/cli/engine/ansible/AnsibleRunner.py [hunk 4]
```diff
@@ -69,13 +68,62 @@ def run(self):
...
+        inventory_upgrade = AnsibleInventoryUpgrade(self.build_dir, self.backup_build_dir)
+        inventory_upgrade.upgrade()
+
+        # generate vars
+        ansible_vars_generator = AnsibleVarsGenerator(inventory_upgrade=inventory_upgrade)      
+        ansible_vars_generator.generate()        
+
+        # pre-flight to prepare machines
+        self.pre_flight(inventory_path)
+
```
Comment (to-bar) [inline core/src/epicli/cli/engine/ansible/AnsibleRunner.py]: I think that for upgrade we need also to run `image-registry` role somewhere.

File: core/src/epicli/cli/engine/ansible/AnsibleRunner.py (hunk 2 above)
Comment (erzetpe) [reply core/src/epicli/cli/engine/ansible/AnsibleRunner.py]: This mistake needs to be fixed.

File: core/src/epicli/cli/engine/ansible/AnsibleRunner.py (hunk 1 above)
Comment (erzetpe) [reply core/src/epicli/cli/engine/ansible/AnsibleRunner.py]: This mistake needs to be fixed.

Comment (seriva) [reply core/src/epicli/cli/engine/ansible/AnsibleRunner.py]: Fixed

File: core/src/epicli/cli/engine/ansible/AnsibleRunner.py (hunk 2 above)
Comment (seriva) [reply core/src/epicli/cli/engine/ansible/AnsibleRunner.py]: Fixed

File: core/src/epicli/cli/epicli.py (hunk 3 above)
Comment (seriva) [reply core/src/epicli/cli/epicli.py]: Fixed

File: core/src/epicli/cli/engine/ansible/AnsibleRunner.py (hunk 4 above)
Comment (seriva) [reply core/src/epicli/cli/engine/ansible/AnsibleRunner.py]: Good point. Added.

**Pull request merged**